packages/firebase_core/firebase_core_platform_interface/pigeons/messages.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:pigeon/pigeon.dart';
+
+@ConfigurePigeon(
+  PigeonOptions(
+    dartOut: 'lib/src/pigeon/messages.pigeon.dart',
+    // We export in the lib folder to expose the class to other packages.
+    dartTestOut: 'lib/src/pigeon/test_api.dart',
+    javaOut:
+        '../firebase_core/android/src/main/java/io/flutter/plugins/firebase/core/GeneratedAndroidFirebaseCore.java',
+    javaOptions: JavaOptions(
+      package: 'io.flutter.plugins.firebase.core',
+      className: 'GeneratedAndroidFirebaseCore',
+    ),
+    objcHeaderOut: '../firebase_core/ios/Classes/messages.g.h',
+    objcSourceOut: '../firebase_core/ios/Classes/messages.g.m',
+  ),
+)
+class PigeonFirebaseOptions {
+  PigeonFirebaseOptions({
+    required this.authDomain,
+    required this.measurementId,
+    required this.deepLinkURLScheme,
+    required this.androidClientId,
+    required this.iosClientId,
+    required this.iosBundleId,
+    required this.appGroupId,
+    required this.apiKey,
+    required this.appId,
+    required this.messagingSenderId,
+    required this.projectId,
+    required this.databaseURL,
+    required this.storageBucket,
+    required this.trackingId,
+  });
+
+  final String apiKey;
+
+  final String appId;
+
+  final String messagingSenderId;
+
+  final String projectId;
+
+  final String? authDomain;
+
+  final String? databaseURL;
+
+  final String? storageBucket;
+
+  final String? measurementId;
+
+  final String? trackingId;
+
+  final String? deepLinkURLScheme;
+
+  final String? androidClientId;
+
+  final String? iosClientId;
+
+  final String? iosBundleId;
+
+  final String? appGroupId;
+}
+
+class PigeonInitializeReponse {
```

Comment:
```suggestion
class PigeonInitializeResponse {
```

Plus the other ones below 😅